src/ensembl/production/metadata/api/genome.py

```diff
@@ -249,48 +398,106 @@ def fetch_sequences_by_assembly_accession(
+                - Attribute: An instance of the Attribute class.
+
+        Raises:
+            ValueError: If an exception occurs during the fetch process.
+
+        """
         try:
             genome_select = db.select(
                 Genome,
+                Organism,
                 GenomeDataset,
                 Dataset,
                 DatasetType,
-                DatasetSource
+                DatasetSource,
             ).select_from(Genome) \
+                .join(Organism, Organism.organism_id == Genome.organism_id) \
                 .join(GenomeDataset, Genome.genome_id == GenomeDataset.genome_id) \
                 .join(Dataset, GenomeDataset.dataset_id == Dataset.dataset_id) \
                 .join(DatasetType, Dataset.dataset_type_id == DatasetType.dataset_type_id) \
-                .join(DatasetSource, Dataset.dataset_source_id == DatasetSource.dataset_source_id)
+                .join(DatasetSource, Dataset.dataset_source_id == DatasetSource.dataset_source_id) \
 
             # set default group topic as 'assembly' to fetch unique datasource
             if not dataset_name:
                 dataset_name = "assembly"
 
             genome_id = check_parameter(genome_id)
             genome_uuid = check_parameter(genome_uuid)
+            organism_uuid = check_parameter(organism_uuid)
             dataset_uuid = check_parameter(dataset_uuid)
             dataset_name = check_parameter(dataset_name)
             dataset_source = check_parameter(dataset_source)
+            dataset_type = check_parameter(dataset_type)
 
             if genome_id is not None:
                 genome_select = genome_select.filter(Genome.genome_id.in_(genome_id))
 
             if genome_uuid is not None:
                 genome_select = genome_select.filter(Genome.genome_uuid.in_(genome_uuid))
 
+            if organism_uuid is not None:
+                genome_select = genome_select.filter(Organism.organism_uuid.in_(organism_uuid))
+
             if dataset_uuid is not None:
                 genome_select = genome_select.filter(Dataset.dataset_uuid.in_(dataset_uuid))
 
             if unreleased_datasets:
                 genome_select = genome_select.filter(GenomeDataset.release_id.is_(None)) \
                     .filter(GenomeDataset.is_current == 0)
-            if dataset_name is not None:
+
+            if is_released:
+                # Include release related info only if is_released is True
+                genome_select = genome_select.add_columns(EnsemblRelease, DatasetAttribute, Attribute) \
+                    .join(EnsemblRelease, GenomeDataset.release_id == EnsemblRelease.release_id) \
+                    .join(DatasetAttribute, DatasetAttribute.dataset_id == Dataset.dataset_id) \
+                    .join(Attribute, Attribute.attribute_id == DatasetAttribute.attribute_id)
+
+            if dataset_name is not None and "all" not in dataset_name:
```

Comment:
Why is the "all" here?

Reply:
By default the API `dataset_name="assembly"`; `all` allows us to fetch everything
Please take a look at this discussion for more context: https://github.com/Ensembl/ensembl-metadata-api/commit/a8eb73d7e3d61d84510dc957cf87777f93e4434c#r121262595

Reply:
Ok, if it works, I am happy.